### DISCUSSION TITLE
Is there offline API documentation for axmol?

### DISCUSSION THREAD

**Discussion by Milieth** [axmolengine/axmol#2245] [Q&A] [answered]
Hi, can I see axmol API  documentation offline ? and where I can download it? cause I can't find it. 
this api [https://axmol.dev/manual/2.2/](https://axmol.dev/manual/2.2/) only available online.
if available that would be nice, thanks

**Comment by rh101** [comment]
> Hi, can I see axmol API documentation offline ? and where I can download it? cause I can't find it.

As far as I know, the API guide has no extra info than what is available in the actual source code header files (it's generated from the source), so you could just clone the Axmol engine source, and you would technically have all the documentation.  If you're not familiar with the functionality in Axmol, then the included `cpp-tests` project is what you should build and run, since it will have an example of most features of the engine. You can then reference the code for any tests you are interested in to see how they are written.

The wiki has some guides, and if you want to read it offline, then you can clone it to your local environment using the following command:

`git clone https://github.com/axmolengine/axmol.wiki.git`

You should then have all the content of the wiki locally, and you will need some kind of markdown reader to so that the content is formatted correctly.  Try something like [md-reader](https://github.com/md-reader/md-reader) for this purpose.

**Reply by Milieth** [reply]
Hi rh101, thanks for your quick reply.
I have do the clone, and now I can see axmol wiki offline (^_^)b.
but then, I still can't open the api [https://axmol.dev/manual/2.2/](https://axmol.dev/manual/2.2/) as offline view, just like I do with cocos2d-x api. Could you help me to acquire it? thanks.

As for the reason, I prefer see it in html format is because I can easily to navigate to a func or class I want to find (the tree view and search engine help me a lot). moreover I can easily see the inherited func and atribute. and lastly sometime the internet not available here.

**Reply by rh101** [reply]
Download and install [doxygen](https://www.doxygen.nl/download.html).

Open a console window, then navigate to the `{axmol}/docs` folder, and run the following command:

`doxygen .\Doxyfile.in`

This will generate the HTML documentation in the `{axmol}/docs/dist` folder.

**Reply by halx99** [reply, marked as answer]
I guess you can also download via `git clone --branch gh-pages https://github.com/axmolengine/axmolengine.github.io`

**Reply by Milieth** [reply]
Thanks guys for the help.
